api/src/main/java/org/openmrs/module/clientregistry/ClientRegistryActivator.java

```diff
@@ -12,26 +11,91 @@
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.openmrs.module.BaseModuleActivator;
+import org.openmrs.module.DaemonToken;
+import org.openmrs.module.DaemonTokenAware;
+import org.openmrs.module.ModuleActivator;
+import org.openmrs.module.clientregistry.api.ClientRegistryManager;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.stereotype.Component;
 
 /**
  * This class contains the logic that is run every time this module is either started or shutdown
  */
-public class ClientRegistryActivator extends BaseModuleActivator {
+
+@Component
+public class ClientRegistryActivator extends BaseModuleActivator implements ApplicationContextAware, DaemonTokenAware {
+	
+	private static ApplicationContext applicationContext;
+	
+	private static DaemonToken daemonToken;
 	
+	// Log
 	private Log log = LogFactory.getLog(this.getClass());
 	
+	@Autowired
+	private ClientRegistryManager clientRegistryManager;
+	
+	@Autowired
+	private ClientRegistryConfig config;
+	
+	/**
+	 * @see ModuleActivator#willStart()
+	 */
+	public void willStart() {
+		log.info("Starting Client Registry Module");
+	}
+	
 	/**
-	 * @see #started()
+	 * @see ModuleActivator#started()
 	 */
 	public void started() {
-		log.info("Started Client Registry");
+		try {
+			applicationContext.getAutowireCapableBeanFactory().autowireBean(this);
+			
+			if (daemonToken != null) {
+				clientRegistryManager.setDaemonToken(daemonToken);
+			}
+			
+			if (config.isClientRegistryEnabled()) {
+				clientRegistryManager.enableClientRegistry();
+			}
+			
+			log.info("Client Registry Module started");
+		}
+		catch (Exception e) {
+			log.error(e.getMessage());
+		}
+		
 	}
 	
 	/**
-	 * @see #shutdown()
+	 * @see ModuleActivator#willStop()
 	 */
-	public void shutdown() {
-		log.info("Shutdown Client Registry");
+	public void willStop() {
+		log.info("Stopping Client Registry Module");
+	}
+	
+	/**
+	 * @see ModuleActivator#stopped()
+	 */
+	public void stopped() {
+		if (clientRegistryManager != null) {
+			clientRegistryManager.disableClientRegistry();
+		}
+		log.info("Client Registry Module stopped");
+	}
+	
+	@Override
+	public void setDaemonToken(DaemonToken token) {
+		this.daemonToken = token;
+	}
+	
+	@Override
+	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+		this.applicationContext = applicationContext;
```

Comment:
applicationContext is static, can we access it via ClientRegistryActivator.applicationContext instead?

Reply:
We need this class to implement ApplicationContextAware to autowire the bean when the module is started